Add 'Block' to the object support table

mitchnielsen opened the pull request:
Adds 'Block' to the object support table. We can mark 'Datasource support' as complete when https://github.com/PrefectHQ/terraform-provider-prefect/issues/175 is complete.

mitchnielsen committed:
Add 'Block' to the object support table

Adds 'Block' to the object support table. We can mark 'Datasource
support' as complete when https://github.com/PrefectHQ/terraform-provider-prefect/issues/175
is complete.

mitchnielsen commented on README.md:
```diff
@@ -30,6 +30,7 @@ Check back with us to see new additions and improvements - and please don't hesi
 | Account Member       |       &check;       |                   |                 |
 | Account Role         |       &check;       |                   |                 |
 | Account              |       &check;       |      &check;      |     &check;     |
+| Block                |                     |      &check;      |     &check;     |
```
I'm going to try to find a list of objects in Prefect, ideally one that delineates which are Cloud-only. We may want to make that distinction somewhere in our TF provider docs.

parkedwards replied:
hah i was just thinking that. from this list, i think we can say:

Cloud-only:
- Account Member
- Account Role
- Account
- Service Account
- Team
- Workspace Access
- Workspace Role
- Workspace

Cloud and OSS
- Block
- Variable
- Work Pool

mitchnielsen replied:
Sweet, thanks - updated in 4daaabc.

mitchnielsen committed:
Add 'Cloud only?' column

Adds a 'Cloud only?' column to the Prefect object table, identifying
which objects are Cloud-only.
